bleak/backends/bluezdbus/manager.py

```diff
@@ -471,10 +471,46 @@ async def passive_scan(
 
                 assert_reply(reply)
 
-                # It is important to export after registering, otherwise BlueZ
-                # won't use the monitor
+                # It is important to export AFTER registering, otherwise BlueZ won't use the monitor
                 self._bus.export(monitor_path, monitor)
 
+                # During export, MGMT command "Add Advertisement Patterns Monitor (0x0052)" is executed.
+                # If Linux kernel is too old to support this command, MGMT event "Command Status (0x0002)"
+                # is returned with "Status: Unknown Command (0x01)" - bleak cannot detect this error and
+                # trying to scan would result in no devices ever reported, as Adv. Monitor was not registered.
+                # Running bluetoothd with -d flag gives the following output for 'journalctl --unit=bluetooth -f':
+                #   bluetoothd[...]: Path /org/bleak/.../AdvertisementMonitor... reserved for Adv Monitor app :...
+                #   ...
+                #   bluetoothd[...]: src/shared/mgmt.c:send_request() [0x0000] command 0x0052
+                #   bluetoothd[...]: src/shared/mgmt.c:can_read_data() [0x0000] command 0x52 status: 0x01
+                #   bluetoothd[...]: Failed to Add Adv Patterns Monitor with status 0x01
+                #   bluetoothd[...]: src/adv_monitor.c:monitor_release() Calling Release() on Adv Monitor
+                #                    of owner :... at path /org/bleak/.../AdvertisementMonitor...
+                #   bluetoothd[...]: Merged_pattern not found when removing monitor
+                #   ...
+                # Note that `monitor.Release()` cannot be invoked if control is not passed to the event loop.
+                # Instead of using asyncio.sleep(), read property to ensure that all queued calls are invoked.
+                reply = await self._bus.call(
+                    Message(
+                        destination=defs.BLUEZ_SERVICE,
+                        path=adapter_path,
+                        interface=defs.PROPERTIES_INTERFACE,
+                        member="Get",
+                        signature="ss",
+                        body=[
+                            defs.ADVERTISEMENT_MONITOR_MANAGER_INTERFACE,
+                            "SupportedFeatures",
+                        ],
+                    )
+                )
```

Comment:
Another option would be to execute
```python
                reply = await self._bus.call(
                    Message(
                        destination=defs.BLUEZ_SERVICE,
                        path=monitor_path,
                        interface=defs.PROPERTIES_INTERFACE,
                        member="Get",
                        signature="ss",
                        body=[defs.ADVERTISEMENT_MONITOR_INTERFACE, "Type"],
                    )
                )
```
however this fails with `['Method "Get" with signature "ss" on interface "org.freedesktop.DBus.Properties" doesn\'t exist\n']`. Considering that even `await asyncio.sleep(0.1)` was sufficient here, I left this call which always succeeds.

Reply:
Tested, and it is not enough. I think that `await asyncio.sleep(0)` does yield, but it does not assure that the D-Bus event/operation queue is completely processed, or processed at all. Just for curiosity, `asyncio.sleep(0.001)` "always" does the job, `0.0001` does the job in about half of the tries, `0.00001` is never enough.